client/utils/uuid.c

```diff
@@ -0,0 +1,100 @@
+//
+//  uuid.c
+//  xdag
+//
+//  Created by Rui Xie on 12/26/18.
+//  Copyright © 2018 xrdavies. All rights reserved.
+//
+
+#include "uuid.h"
+#include <stdint.h>
+#include <stdlib.h>
+
+#if defined(_WIN32)
+#include <windows.h>
+#include <wincrypt.h>
+#endif
+
+
+static const char *template = "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx";
+static const char *chars = "0123456789abcdef";
+
+/* http://xorshift.di.unimi.it/xorshift128plus.c */
+static uint64_t xorshift128plus(uint64_t *s) {
+	uint64_t s1 = s[0];
+	uint64_t s0 = s[1];
+	s[0] = s0;
+	s1 ^= s1 << 23;
+	s[1] = s1 ^ s0 ^ (s1 >> 18) ^ (s0 >> 5);
+	return s[1] + s0;
+}
+
+int uuid(char *buf)
```

Comment:
cannot reuse our PRG?

Reply:
Yes, good point. Meant to reuse our RPG, but when I implemented this code, we still donot have RPG yet.

Reply:
Will try to replace it later.